### PR TITLE
dsl: support for js-dsl

### PR DESCRIPTION
**Pull request by lburgazzoli**
@jamesnetherton @ppalaga mind having a look at this pr ? 

native test are likely for fail because of the compilation requires ~12 GB

### CODE REVIEW

**Comment by ppalaga** [comment]
I'd have a look tomorrow, depending on how well https://github.com/apache/camel-quarkus/pull/2371 is going to pass

File: docs/modules/ROOT/pages/reference/extensions/js-dsl.adoc [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// Do not edit directly!
+// This file was generated by camel-quarkus-maven-plugin:update-extension-doc-page
+= JavaScript DSL
+:cq-artifact-id: camel-quarkus-js-dsl
+:cq-native-supported: true
+:cq-status: Stable
+:cq-description: An YAML stack for parsing YAML route definitions
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/reference/extensions/js-dsl.adoc:7]: `An YAML stack for parsing YAML route definitions` does not sound right.

Comment (lburgazzoli) [reply docs/modules/ROOT/pages/reference/extensions/js-dsl.adoc:7]: yah, forgot to regen

**Comment by ppalaga** [comment]
Would you mind rebasing on top of https://github.com/apache/camel-quarkus/pull/2371 ?

**Comment by ppalaga** [comment]
Superseded by https://github.com/apache/camel-quarkus/pull/2385

**Pull request closed without merging**